pkg/crc/cluster/cluster.go

```diff
@@ -168,18 +168,31 @@ func RemovePullSecretFromCluster(ocConfig oc.Config, sshRunner *ssh.Runner) erro
 		return fmt.Errorf("Failed to remove Pull secret %w: %s", err, stderr)
 	}
 
-	cmdArgs = []string{"delete", "machineconfigs", "--all"}
-	_, stderr, err = ocConfig.RunOcCommand(cmdArgs...)
+	return nil
+}
+
+func RemoveOldRenderedMachineConfig(runner *ssh.Runner) error {
+	// This block (#L179-183) should be removed as soon as we start shipping with 4.8 bundle.
+	// This check is only make sure if there is any machineconfig resource or not because
+	// in 4.7 we disabled mco and also deleted all the machineconfig/machineconfig-pools.
+	// For 4.8 we don't disable mco and it does contain the machineconfigs.
+	cmd := "oc --kubeconfig=/opt/kubeconfig get mc --sort-by=.metadata.creationTimestamp --no-headers -oname | grep rendered-master"
+	_, _, err := runner.Run(cmd)
+	if err != nil {
+		return nil
+	}
+
+	cmd = "oc --kubeconfig=/opt/kubeconfig get mc --sort-by=.metadata.creationTimestamp --no-headers -oname | grep rendered-master | head -n -1 | xargs -t oc --kubeconfig=/opt/kubeconfig delete"
+	_, stderr, err := runner.Run(cmd)
```

Comment:
Could these various `oc` commandlines make use of `OcRunner`/`OcConfig`?

Reply:
@cfergeau `OcRunner` doesn't support pipe so if we really go with `OcRunner` then we need to do something like following.
```
mc, stderr, err := ocConfig.RunOcCommand("get mc --sort-by=.metadata.creationTimestamp --no-headers -oname")
[...]
oldmc, stderr, err := sshRunner.Run(fmt.Sprintf("%s | grep rendered-master | head -n -1", mc)
[...]
_, stderr,err := ocConfig.RunOcCommand(fmt.Sprintf("delete %s", oldmc))
```

Reply:
We don't have to use `| grep| head` to achieve this, we could ask for json output and let go parse it (do we already vendor the module with the type definitions?), or use https://golang.org/pkg/bufio/#NewScanner to get what we are interested in?

Reply:
@cfergeau I will check again with json output.